packages/ai-jsx/src/batteries/sidekick/platform/conversation.tsx

```diff
@@ -136,9 +161,33 @@ export function getNextConversationStep(
     case 'system':
     case 'user':
     case 'functionResponse': {
-      const generation = (
-        <ChatCompletion functionDefinitions={tools ? updatedTools : undefined}>{shrinkableConversation}</ChatCompletion>
+      let generation = (
```

Comment:
Q: shouldn't we do this instead as soon as RAG is requested not when we are trying to respond to RAG, i.e. when trying to execute the function?
Or am I missing something?

Reply:
Yes, exactly -- note that this is wrapping the `<ChatCompletion>` (which produces `<FunctionCall>`s), not `<ExecuteFunction>` (which produces `<FunctionResponse>`s).

So as soon as we see that the `<ChatCompletion>` is _requesting_ a function call, the content will be interjected.

Reply:
Okay I understand the issue now. I was reading the code as `if case 'functionResponse' then do X`, but it was actually `if case system/user/response then do X`.